src/main/java/org/jenkinsci/plugins/workflow/graphanalysis/AbstractFlowScanner.java

```diff
@@ -244,15 +247,16 @@ public final void remove() {
      * @return First matching node, or null if no matches found
      */
     @CheckForNull
+    @Deprecated
     public FlowNode findFirstMatch(@CheckForNull Collection<FlowNode> heads,
-                                           @CheckForNull Collection<FlowNode> blackListNodes,
-                                           Predicate<FlowNode> matchCondition) {
+                                   @CheckForNull Collection<FlowNode> blackListNodes,
+                                   Predicate<FlowNode> matchCondition) {
```

Comment:
FWIW, I think changing signatures like this will be a breaking change for some plugins.

Reply:
Yup agree, it's a bit too agressive change... Even if I checked with usage-in-plugins and there is not so much usage of that.
If the code is using only lambda is not a problem it will work with this change :)
but yeah might be better to keep both (but that's a lot... especially a lot are already marked deprecated..)

Reply:
Yeah, I think a lot of cases would be fine, but stuff like [this](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/15978cd172a8d34e2cd6bf8a6bfeaa5514d44dad/pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy#L158-L176) along with callers like [this](https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/4e07aed2384229428debc93da38512ddc60fdf60/pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy#L358) would need to be updated.